conda_env/specs/notebook.py

```diff
@@ -0,0 +1,35 @@
+try:
+    from IPython import nbformat
+except ImportError:
+    nbformat = None
+from ..env import Environment
+from .binstar import BinstarSpec
+
+
+class NotebookSpec(object):
+    msg = None
+
+    def __init__(self, name=None, **kwargs):
+        self.name = name
+        self.nb = {}
+
+    def can_handle(self):
+        try:
+            self.nb = nbformat.reader.reads(open(self.name).read())
+            return 'environment' in self.nb['metadata']
+        except AttributeError:
+            self.msg = "Please install IPython notebook"
+        except IOError:
+            self.msg = "{} does not exist o can't be accessed".format(self.name)
+        except (nbformat.reader.NotJSONError, KeyError) as e:
+            print(e)
+            self.msg = "{} does not looks like a notebook file".format(self.name)
+        return False
```

Comment:
I would expect a function that returns False would also be capable of returning True.  Maybe return None if unsuccessful?  In that case I would rename the function get_env() or something similar which returns the environment if available, otherwise None.